src/foam/nanos/dig/SugarWebAgent.java

```diff
@@ -149,9 +149,8 @@ public void execute(X x) {
           }
           executeMethod(x, resp, out, class_, serviceName, methodName, paramTypes, arglist);
         }
-        if ( ! methodFound ) throw new RuntimeException("Invalid method: " + methodName);
       }
-
+      if ( ! methodFound ) throw new RuntimeException("Invalid method: " + methodName);
```

Comment:
Best not to include user input in error messages. Otherwise security auditors will flag it as a security issue. For example: methodName=<script>txnDAO.put(Transaction.create({send money to a bad guy}));</script>

Reply:
Just say "Invalid Method" or "Unknown Method".